x/pool/README.md

```diff
@@ -0,0 +1,6 @@
+---
+sidebar_position: 1
+---
+
+# `x/pool`
```

Comment:
thoughts on naming it protocol-pool?

Reply:
protocol-pool does seem more sensible and shows module's purpose. 
I changed usage of `ModuleName` to this. My inclination is to keep module package name as `x/pool` as it is shorter, easier to use